pkg/cloud/openstack/services/userdata/kubeadm.go

```diff
@@ -0,0 +1,164 @@
+package userdata
+
+import (
+	"fmt"
+	"k8s.io/klog"
+	kubeadmv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
+	providerv1 "sigs.k8s.io/cluster-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
+	"sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/services/certificates"
+	"sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/services/kubeadm"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+const (
+	localIPV4Lookup = "${COREOS_OPENSTACK_IPV4_LOCAL}"
```

Comment:
Where does this value come from? `"${COREOS_OPENSTACK_IPV4_LOCAL}"` and why is it specific to coreos?

Reply:
> thanks a lot for the PR :)
> I am still struggling with the LB issue so I can't test this locally with LB, just to confirm, we support ubuntu, rhel and coreos previous, with this change all those will still be supported (local tested and future we will have CI) ,right?

Yup. The idea at least mine. Is to support the same operating systems for now. But somebody has to help me test :).

I didn't try it but the goal should be that this PR also works without APIServer LoadBalancer and single-node master.

If you want we can schedule a meeting, and I can take a look at your LB issue.

Reply:
That's just an adapted version of the CAPA code. They used a similar constant of cloud-init :/ 
But yes you're right we have to find a way to get the local IP independent of the OS. This variable is currently replaced through the real IP by the `/op/bin/prepare.sh` script from user-data

Reply:
I'll just revert to the old scripting in the userdata file and use some other name for the var